src/Silverback.Core/Diagnostics/ILogLevelDictionary.cs

```diff
@@ -10,7 +10,7 @@ namespace Silverback.Diagnostics
     /// <summary>
     ///     Maps <see cref="EventId" />s to the <see cref="LogLevel" /> that should be used for it.
     /// </summary>
-    public interface ILogLevelDictionary : IReadOnlyDictionary<EventId, Func<Exception, LogLevel, LogLevel>>
+    public interface ILogLevelDictionary : IReadOnlyDictionary<EventId, Func<Exception, LogLevel, Lazy<string>, LogLevel>>
```

Comment:
I use a `Lazy<string>` to avoid the calculation of the message if it's not explicitly needed by the log level configuration function. Because logging might be contained in hot paths this seems a good idea to me to avoid a performance impact.